nodejs/week1/homework/src/backend/app.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+// import data here
+const meals = require('./data/meals');
+const reviews = require('./data/reviews');
+const reservations = require('./data/reservations');
+const colours = require('colors/safe');
+// this is where you will be adding your routes
+app.get('/', async(request, response) => {
+    // response.send("asd");
+    console.log(colours.rainbow('hello this is my first node message'));
+    response.send(colours.green('hello this is my first node message'));
+});
+
+// Respond with the json for all the meals.
+// For each meal, if there are reviews matching it's meal ID,
+// add these reviews to each meal in the form of an array.
+// For meals that do not have any reviews, the "reviews" property will be an empty array.
+
+app.get('/meals', async(request, response) => {
+    const mealsWithReviews = meals.map((meal) => {
+        const reviewsForMeal = reviews.filter(
+            (review) => review.mealId === meal.id,
+        );
+        return {
+            ...meal,
+            reviews: reviewsForMeal,
+        };
+    });
+    response.send(mealsWithReviews);
+
+    // response.json(mealsWithReviews);
```

Comment:
In this case both work but it's preferred to use .json -> https://hackyourfuture-cph.slack.com/archives/C02EXR39E5R/p1645710223738659?thread_ts=1645709966.699389&cid=C02EXR39E5R

Reply:
could not open this link